manager/dispatcher/attachment.go

```diff
@@ -0,0 +1,75 @@
+package dispatcher
+
+import (
+	"github.com/docker/swarmkit/api"
+	//"github.com/docker/swarmkit/ca"
+	"github.com/docker/swarmkit/identity"
+	"github.com/docker/swarmkit/log"
+	"github.com/docker/swarmkit/manager/state/store"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+)
+
+// CreateNetworkAttachment allows the node to request the resources
+// allocation needed for a network attachment on the specific node.
+// - Returns `InvalidArgument` if the Spec is malformed.
+// - Returns `NotFound` if the Network is not found.
+// - Returns an error if the creation fails.
+func (d *Dispatcher) CreateNetworkAttachment(ctx context.Context, request *api.CreateNetworkAttachmentRequest) (*api.CreateNetworkAttachmentResponse, error) {
+	att := &api.NetworkAttachment{
+		ID:     identity.NewID(),
+		NodeID: request.Config.NodeID,
+		Config: *request.Config,
+	}
+
+	d.store.View(func(tx store.ReadTx) {
+		att.Network = store.GetNetwork(tx, att.Config.Target)
+		if att.Network == nil {
+			if networks, err := store.FindNetworks(tx, store.ByName(att.Config.Target)); err == nil && len(networks) > 0 {
+				att.Network = networks[0]
+			}
+		}
+	})
+	if att.Network == nil {
+		return nil, grpc.Errorf(codes.NotFound, "network  %s not found", att.Config.Target)
+	}
+
+	for _, addr := range att.Config.Addresses {
+		att.Addresses = append(att.Addresses, addr)
+	}
+
+	if err := d.store.Update(func(tx store.Tx) error {
+		return store.CreateAttachment(tx, att)
+	}); err != nil {
+		return nil, err
+	}
+
+	log.G(ctx).Debugf("Executor Attachment %s created and saved to store", att.ID)
+
+	return &api.CreateNetworkAttachmentResponse{ID: att.ID}, nil
+}
+
+// RemoveNetworkAttachment allows the node to request the release of
+// the resources associated to the network attachment.
+// - Returns `InvalidArgument` if attachment ID is not provided.
+// - Returns `NotFound` if the attachment is not found.
+// - Returns an error if the deletion fails.
+func (d *Dispatcher) RemoveNetworkAttachment(ctx context.Context, request *api.RemoveNetworkAttachmentRequest) (*api.RemoveNetworkAttachmentResponse, error) {
```

Comment:
Thanks. Will make sure a worker can only delete attachments created on the same node